src/telemetry/stats.c

```diff
@@ -577,13 +584,12 @@ ts_telemetry_stats_gather(TelemetryStats *stats)
 					process_relation(&stats->views, class);
 				break;
 			case RELTYPE_MATVIEW:
-				process_relation(&stats->materialized_views.base, class);
+				process_relation(&stats->materialized_views.bs, class);
 				break;
 			case RELTYPE_CONTINUOUS_AGG:
 				Assert(NULL != cagg);
-				process_continuous_agg(&stats->continuous_aggs.storage.base, class, cagg);
+				process_continuous_agg(&stats->continuous_aggs.hs.ss.bs, class, cagg);
```

Comment:
This seems overly complicated. I am not sure why you're passing in a pointer to the "base" class here instead of just a pointer to the continuous aggs state. Since you have no checks of the accurate type inside the function, this only opens up the window for bugs that cause crashes if a bad pointer is passed in.

```suggestion
				process_continuous_agg(&stats->continuous_aggs, class, cagg);
```

Reply:
The base object is passed on the process_relation() within that function, but I can undwind/cast it there instead if you prefer.

Reply:
I think that is better, yes. It will capture any type issues at compile-time instead of run-time.